src/html.rs

```diff
@@ -89,4 +121,49 @@ mod test {
     fn static_config_parses() {
         assert!(!CONFIG.sections_to_remove.is_empty());
     }
+
+    #[test]
+    fn remove_links() {
+        let html = r#"
+        <p> Some text that includes
+            <a href="Some_Page"><span id="inner-content">several</span></a>
+            <a id="second-link" href="./Another_Page">relative links</a>
+        and
+            <a href="https://example.com/page">an absolute link</a>
+        .
+        </p>
+        "#;
+
+        let anchors = Selector::parse("a").unwrap();
+        let inner_element = Selector::parse("#inner-content").unwrap();
+        let second_link = Selector::parse("#second-link").unwrap();
+
+        let mut document = Html::parse_fragment(html);
+        let links: Vec<_> = document
+            .select(&anchors)
+            .filter_map(|el| el.value().attr("href"))
+            .collect();
+
+        eprintln!("{}", document.html());
+
+        assert_eq!(
+            vec!["Some_Page", "./Another_Page", "https://example.com/page"],
+            links
+        );
+
+        // orphan one of the links
```

Comment:
Can this comment be clarified?

Reply:
```suggestion
        // Detach one of the links from the root tree (as if previously deleted) to ensure it handles orphan nodes nicely.
```
Some of the tree operations [panic when the node doesn't have a parent](https://docs.rs/ego-tree/latest/ego_tree/struct.NodeMut.html#method.insert_id_before), and we could be processing nodes that were removed from the tree in a previous pass.